LibOS/shim/src/shim_malloc.c

```diff
@@ -360,20 +362,27 @@ void * calloc (size_t nmemb, size_t size)
 }
 extern_alias(calloc);
 
-void* realloc(void* ptr, size_t new_size)
+#if 0 /* Temporarily disabling this code */
+void * realloc(void * ptr, size_t new_size)
 {
-    // TODO: This function should be optimized to avoid memcpy when buffer
-    // expansion is possible.
+    /* TODO: We can't deal with this case right now */
+    assert(!MEMORY_MIGRATED(ptr));
+
     size_t old_size = slab_get_buf_size(slab_mgr, ptr);
-    void* new_buf = malloc(new_size);
+    if (old_size >= new_size)
+        return ptr;
+
+    void * new_buf = malloc(new_size);
     if (!new_buf)
         return NULL;
-    memcpy(new_buf, ptr, new_size < old_size ? new_size : old_size);
+
+    memcpy(new_buf, ptr, old_size);
+    memset(new_buf + old_size, 0, new_size - old_size);
```

Comment:
This zeroing isn't a part of the original `realloc` semantics, the caller can't rely upon it. Please either remove it or if it's supposed to be a debugging canary, then change it to something like "0xCC" and ifdef it to be enabled in debug mode only.